### PR TITLE
refactor: use the original from/to values when creating the bucket

### PR DESCRIPTION
**Pull request by salvatore-campagna**
The range bucket values should always be set to the original `from`
and `to` values. These values are the ones used by the serialisation
logic and are the values the client expects in the response.

Also, the key for a Range is made optional, avoiding unnecessary
serialisation and deserialisation. The actual key generation
happens only if required (calling  'getKey' or 'getKeyAsString'
directly or at key serialisation time).
A Range does not actually need a key to accomplish its purpose ('from'
and 'to' are the only required parameters), other than propagating a
user-specified key for the bucket collecting documents for that range.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/RangeAggregator.java [hunk 1]
```diff
@@ -121,22 +124,33 @@
...
          * )})
          */
-        public Range(String key, Double from, Double originalFrom, String fromAsStr, Double to, Double originalTo, String toAsStr) {
+        public Range(
+            final String key,
+            final Double from,
+            final String fromAsStr,
+            final Double to,
+            final String toAsStr,
+            final DoubleUnaryOperator op
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/RangeAggregator.java:133]: Rename `op` to `fixPrecision` I think

**Review by nik9000 — COMMENTED** [review]
OH! That's better.

**Review by nik9000 — APPROVED** [review]

**Comment by salvatore-campagna** [comment]
In order to fix the BWC issue I would need this patch to "overwrite" the patch introduced with #81801.
#81801 was back-ported to `7.17.0` and `8.0.0`.

Right now this is failing because older versions (`7.17.0` and `8.0.0` expect the `originalFrom` and `originalTo` values to be serialised into the wire format. I think after overwriting the previous PR with this one in `7.17.0` and `8.0.0` the BWC tests for this PR will go green.

**Comment by salvatore-campagna** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|7.17|[](https://github.com/elastic/elasticsearch/pull/83339)|


### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**